openedx/core/djangoapps/kafka_consumer/management/commands/consume_events.py

```diff
@@ -0,0 +1,94 @@
+from confluent_kafka.serialization import StringDeserializer
+from openedx.core.djangoapps.kafka_consumer.consumers import getHandler
+
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Listen for license events from the event bus and log them. Only run on servers where KAFKA_ENABLED is true
+    """
+    help = """
+    This starts a Kafka event consumer that listens to the specified topic and logs all messages it receives. Topic
+    is required.
+
+    example:
+        manage.py ... consume_events -t license-event-prod
+
+    """
+
+    def add_arguments(self, parser):
+
+        parser.add_argument(
+            '-t', '--topic',
+            nargs=1,
+            required=True,
+            help='Topic to consume'
+        )
+
+        parser.add_argument(
+            '-g', '--group_id',
+            nargs=1,
+            required=True,
+            help='Consumer group id'
+        )
+
+    def handle(self, *args, **options):
+        if not settings.KAFKA_ENABLED:
+            logger.error("Kafka not enabled")
+            return
+        try:
+            KAFKA_SCHEMA_REGISTRY_CONFIG = {
+                'url': settings.SCHEMA_REGISTRY_URL,
+                'basic.auth.user.info': f"{settings.SCHEMA_REGISTRY_API_KEY}"
+                f":{settings.SCHEMA_REGISTRY_API_SECRET}",
+            }
+
+            schema_registry_client = SchemaRegistryClient(KAFKA_SCHEMA_REGISTRY_CONFIG)
+
+            topic = options['topic'][0]
+
+            HandlerClass = getHandler(topic)
+
+            consumer = DeserializingConsumer({
+                'bootstrap.servers': settings.KAFKA_BOOTSTRAP_SERVER,
+                'group.id': options["group_id"][0],
+                'key.deserializer': StringDeserializer('utf-8'),
+                'value.deserializer': HandlerClass.getDeserializer(schema_registry_client),
+                'auto.offset.reset': 'earliest',
+                'sasl.mechanism': 'PLAIN',
+                'security.protocol': 'SASL_SSL',
+                'sasl.username': settings.KAFKA_API_KEY,
+                'sasl.password': settings.KAFKA_API_SECRET,
+            })
+
+            try:
+                consumer.subscribe([topic])
+
+                while True:
+                    msg = consumer.poll(timeout=1.0)
+                    if msg is None:
+                        continue
+                    if msg.error():
+                        if msg.error().code() == KafkaError._PARTITION_EOF:  # pylint: disable=protected-access
+                            # End of partition event
+                            logger.info(f"{msg.topic()} [{msg.partition()}] reached end at offset {msg.offset}")
+                        elif msg.error():
+                            logger.exception(msg.error())
+                    else:
+                        HandlerClass.handleMessage(msg)
+            finally:
+                # Close down consumer to commit final offsets.
+                consumer.close()
```

Comment:
1. Does this get run if you CTRL-C the process locally in attached mode? How do we properly shutdown one of these processes? Do we need to be able to signal it to break out of the loop and close? Are these questions for the future?
2. Could this fail if the consumer was never set up appropriately? Does this need its own try block to log the failure?

Reply:
1. I'm leaving that question to the future since I think that's going to depend on infrastructure work. I've confirmed the finally block does get run with a CTRL-C exit. I'll put a TODO in there

2. Local testing suggests this will be fine even if the consumer is misconfigured. The documentation on the confluent code suggests that this really shouldn't raise anything although I'm not overly familiar with translating C to py https://github.com/confluentinc/confluent-kafka-python/blob/e671bccb8a4f98302748ccf60d5d579f68c6613d/src/confluent_kafka/src/Consumer.c#L1412